lib/portage/tests/news/test_NewsItem.py

```diff
@@ -1,109 +1,347 @@
+            msg=f"Expected {item} to be relevant, but it was not!",
+        )
+
+        # What if we have a newer version of the listed package?
+        self.vardb.cpv_inject("net-misc/openssh-9.2_p1", {"SLOT": "0"})
+        item = self._createNewsItem(
+            {
+                "display_if_installed": [
+                    "<net-misc/openssh-9.2_p1",
+                ]
+            }
+        )
+        self.assertTrue(item.isValid())
+        self.assertFalse(
+            item.isRelevant(self.vardb, self.settings, self.profile),
+            msg=f"Expected {item} to be irrelevant, but it was relevant!",
+        )
 
     def testDisplayIfKeyword(self):
-        tmpItem = self.fakeItem[:].replace(
-            "#Display-If-Keyword:", f"Display-If-Keyword: {self.keywords}"
+        item = self._createNewsItem({"display_if_keyword": [self.keywords]})
+        self.assertTrue(item.isValid())
+        self.assertTrue(
+            item.isRelevant(self.vardb, self.settings, self.profile),
+            msg=f"Expected {item} to be relevant, but it was not!",
         )
 
-        try:
-            item = self._processItem(tmpItem)
-            self.assertTrue(
-                item.isRelevant(self.vardb, self.settings, self.profile),
-                msg=f"Expected {tmpItem} to be relevant, but it was not!",
-            )
-        finally:
-            os.unlink(item.path)
-
-    def _processItem(self, item):
-        filename = None
-        fd, filename = mkstemp()
-        f = os.fdopen(fd, "w")
-        f.write(item)
-        f.close()
-        try:
-            return NewsItem(filename, 0)
-        except TypeError:
-            self.fail(f"Error while processing news item {filename}")
+        # Test the negative case: a keyword we don't have set.
+        item = self._createNewsItem({"display_if_keyword": ["fake-keyword"]})
+        self.assertTrue(item.isValid())
+        self.assertFalse(
+            item.isRelevant(self.vardb, self.settings, self.profile),
+            msg=f"Expected {item} to be irrelevant, but it was relevant!",
+        )
+
+        # What if several keywords are listed and we match one of them?
+        item = self._createNewsItem(
+            {"display_if_keyword": [self.keywords, "amd64", "~hppa"]}
+        )
+        self.assertTrue(item.isValid())
+        self.assertTrue(
+            item.isRelevant(self.vardb, self.settings, self.profile),
+            msg=f"Expected {item} to be relevant, but it was not!",
+        )
+
+        # What if several keywords are listed and we match none of them?
+        item = self._createNewsItem({"display_if_keyword": ["amd64", "~hppa"]})
+        self.assertTrue(item.isValid())
+        self.assertFalse(
+            item.isRelevant(self.vardb, self.settings, self.profile),
+            msg=f"Expected {item} to be irrelevant, but it was relevant!",
+        )
+
+        # What if the ~keyword (testing) keyword is listed but we're keyword (stable)?
```

Comment:
Maybe it's worth it to also add the case in which we're testing and the stable keyword is listed in the news item?

Reply:
Good idea, done!